hack/update-vendor-licenses.sh

```diff
@@ -186,6 +186,45 @@ if [ -f "${LICENSE_ROOT}/LICENSE" ]; then
   mv "${TMP_LICENSE_FILE}" "${TMP_LICENSES_DIR}/LICENSE"
 fi
 
+# only_contains_submodules will check whether a package is empty of files
+# and its subdirectories are all modules, which indicates we do not need
+# a LICENSE for this module as it is an indirect transitive dependency.
+only_contains_submodules () {
+  package="$1"
+  if [[ -z "$(find "${DEPS_DIR}/${package}/" -mindepth 1 -maxdepth 1 -type f)" ]]; then
+    # If the package does not contain files, check whether all subdirectories
+    # only contain modules.
+    while read -d "" -r SUBDIR; do
+      # Because we have a trailing '/' on the 'find' command's argument, we
+      # must strip off two '/' from the end of the package path before the
+      # sub-packages name.
+      subpackage="${package}/${SUBDIR#"${DEPS_DIR}/${package}//"}"
+
+      # If the subdirectory is already a module, don't recurse into deeper
+      # subdirectories of this subpackage.
+      if go list -m "${subpackage}"; then
```

Comment:
This returns "not a known dependency" (with a non-zero exit code) if the given package is not a module, so it's okay to do the check without matching against the output of `go list -m all`